### ISSUE TITLE
UPDATE crashes on non-indexed predicates

### ISSUE THREAD

**Issue by PauloAmora** [cmu-db/peloton#545]
The UPDATE command crashes if used with a non-index predicate or without one.

Start by creating a table and adding some data:
```SQL
CREATE TABLE foo (a INTEGER, b INTEGER);
INSERT INTO foo VALUES (1, 200);
INSERT INTO foo VALUES (2, 150);
INSERT INTO foo VALUES (3, 400);
```

If you execute the following command:
```SQL
UPDATE foo SET b = 100 WHERE a = 2;
```

The system crashes. It also happens with queries like this:
```SQL
UPDATE foo SET b = 100;
```

Test script is here: http://pastebin.com/zq7dtPac

**Comment by seojungmin** [comment]
Attached is a quick patch to the issue.  The crash comes from asserting (index_entry_ptr != nullptr).  index_entry_ptr may be equal to nullptr if there is no index.  Since the previous codefor performUpdate() does not align with handling index_entry_ptr in perfromDelete(), I assume that this is a bug. 

[peloton-545.diff.txt](https://github.com/cmu-db/peloton/files/802186/peloton-545.diff.txt)

**Comment by apavlo** [comment]
@seojungmin Can you send a PR? We don't do direct patches.

**Comment by poojanilangekar** [comment]
Fixed in #558
